### PR TITLE
BTFSBTT-201: Modify btfs add command to add chunk metadata

### PR DESCRIPTION
**Pull request by steveyeom**
1. "btfs add" command changes
   core/commands/add.go - options
2. Adding metadata functionality: 
   core/coreapi/unixfs.go - set up passing metadata json string
   core/coreunix/add.go - convert json string to byte array and verify string validity
3. Dependency resolving code changes to import TRON-US repos. 
   All the files but the above files

go.sum will be gone for the next commit. Sorry. 
go fmt and go mod tidy will be performed.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -6,8 +6,11 @@ require (
 	github.com/FactomProject/btcutilecc v0.0.0-20130527213604-d3a63a5752ec // indirect
 	github.com/Kubuxu/gocovmerge v0.0.0-20161216165753-7ecaa51963cd
 	github.com/StackExchange/wmi v0.0.0-20181212234831-e0a55b97c705 // indirect
-	github.com/TRON-US/go-btfs-cmds v0.1.3
+	github.com/TRON-US/go-btfs-cmds v0.1.4
 	github.com/TRON-US/go-btfs-config v0.1.5
+	github.com/TRON-US/go-mfs v0.2.0
+	github.com/TRON-US/go-unixfs v0.3.0
+	github.com/TRON-US/interface-go-btfs-core v0.3.0
```
Comment (taiyangc) [inline go.mod]: I thought you had another update of `interface-go-btfs-core` just now? did you tag a new release?

File: go.mod [hunk 2]
```diff
@@ -25,7 +28,7 @@ require (
 	github.com/ipfs/dir-index-html v1.0.3
 	github.com/ipfs/go-bitswap v0.1.6
 	github.com/ipfs/go-block-format v0.0.2
-	github.com/ipfs/go-blockservice v0.1.0
+	github.com/ipfs/go-blockservice v0.1.1
```
Comment (taiyangc) [inline go.mod]: Is this intentional?

File: go.mod (hunk 1 above)
Comment (steveyeom) [reply go.mod]: Yes. v0.3.1.

File: go.mod (hunk 2 above)
Comment (steveyeom) [reply go.mod]: No

Comment (steveyeom) [reply go.mod]: will change back to v0.1.0

**Comment by steveyeom** [comment]
There were multiple depending repo changes meanwhile. Even though the changes are not much, that complicated a bit.  I will close this pull request and submit another one to make things easy.

**Pull request closed without merging**